test/hermes/text-decoder.js

```diff
@@ -0,0 +1,247 @@
+  // CHECK-NEXT: RangeError
+}
+
+// Test decoding empty input
+print(decoder.decode() === '');
+// CHECK-NEXT: true
+
+print(decoder.decode(undefined) === '');
+// CHECK-NEXT: true
+
+print(decoder.decode(new Uint8Array(0)) === '');
+// CHECK-NEXT: true
+
+// Test basic UTF-8 decoding
+var utf8Bytes = new Uint8Array([116, 101, 115, 116]); // "test"
+print(decoder.decode(utf8Bytes));
+// CHECK-NEXT: test
+
+// Test UTF-8 with multi-byte characters
+// "↑↓" = U+2191 U+2193 = E2 86 91 E2 86 93
+var arrowBytes = new Uint8Array([0xE2, 0x86, 0x91, 0xE2, 0x86, 0x93]);
+print(decoder.decode(arrowBytes));
+// CHECK-NEXT: ↑↓
+
+// Test UTF-8 with 4-byte character (emoji)
+// "😃" = U+1F603 = F0 9F 98 83
+var emojiBytes = new Uint8Array([0xF0, 0x9F, 0x98, 0x83]);
+print(decoder.decode(emojiBytes));
+// CHECK-NEXT: 😃
+
+// Test UTF-8 BOM handling (default: skip BOM)
+var utf8WithBOM = new Uint8Array([0xEF, 0xBB, 0xBF, 116, 101, 115, 116]);
+print(decoder.decode(utf8WithBOM));
+// CHECK-NEXT: test
+
+// Test UTF-8 BOM with ignoreBOM=true (include BOM in output)
+var decoderIgnoreBOM = new TextDecoder('utf-8', { ignoreBOM: true });
+var result = decoderIgnoreBOM.decode(utf8WithBOM);
+print(result.length);
+// CHECK-NEXT: 5
+
+// Test invalid UTF-8 handling (non-fatal, replacement character)
+var invalidUtf8 = new Uint8Array([0x80, 0x81, 116, 101, 115, 116]);
+var decoded = decoder.decode(invalidUtf8);
+print(decoded.length);
+// CHECK-NEXT: 6
+print(decoded.charCodeAt(0));
+// CHECK-NEXT: 65533
+print(decoded.charCodeAt(1));
+// CHECK-NEXT: 65533
+
+// Test fatal mode for invalid UTF-8
+var fatalDecoder = new TextDecoder('utf-8', { fatal: true });
+try {
+  fatalDecoder.decode(new Uint8Array([0x80]));
+} catch (e) {
+  print(e.name);
+  // CHECK-NEXT: TypeError
+}
+
+// Test UTF-16LE decoding
+// "test" in UTF-16LE = 74 00 65 00 73 00 74 00
+var utf16leBytes = new Uint8Array([0x74, 0x00, 0x65, 0x00, 0x73, 0x00, 0x74, 0x00]);
+print(utf16leDecoder.decode(utf16leBytes));
+// CHECK-NEXT: test
+
+// Test UTF-16LE with BOM (FF FE)
+var utf16leWithBOM = new Uint8Array([0xFF, 0xFE, 0x74, 0x00, 0x65, 0x00, 0x73, 0x00, 0x74, 0x00]);
+print(utf16leDecoder.decode(utf16leWithBOM));
+// CHECK-NEXT: test
+
+// Test UTF-16BE decoding
+// "test" in UTF-16BE = 00 74 00 65 00 73 00 74
+var utf16beBytes = new Uint8Array([0x00, 0x74, 0x00, 0x65, 0x00, 0x73, 0x00, 0x74]);
+print(utf16beDecoder.decode(utf16beBytes));
+// CHECK-NEXT: test
+
+// Test UTF-16BE with BOM (FE FF)
+var utf16beWithBOM = new Uint8Array([0xFE, 0xFF, 0x00, 0x74, 0x00, 0x65, 0x00, 0x73, 0x00, 0x74]);
+print(utf16beDecoder.decode(utf16beWithBOM));
+// CHECK-NEXT: test
+
+// Test Latin-1 decoding
+// "café" in Latin-1 = 63 61 66 E9
+var latin1Bytes = new Uint8Array([0x63, 0x61, 0x66, 0xE9]);
+print(latin1Decoder.decode(latin1Bytes));
+// CHECK-NEXT: café
+
+// Test Latin-1 with high bytes
+var latin1High = new Uint8Array([0xA9, 0xAE, 0xB0]); // ©®°
+print(latin1Decoder.decode(latin1High));
+// CHECK-NEXT: ©®°
```

Comment:
There is no Latin1 mode in TextDecoder

There is `windows-1252` which `latin1` aliases to

`new TextDecoder('latin1')` and `new TextDecoder('ascii')` both create a `windows-1252` decoder
Which also could of course decode ascii and latin1 bytes, as those are a _subset_ of `windows-1252`

But it should fully decode windows-1252, even if it was instantiated with `ascii`

To avoid confusion, don't call this a Latin1 mode